src/libpgagroal/management.c

```diff
@@ -489,6 +489,7 @@ pgagroal_management_get_password(SSL* ssl, int fd, char* username, char* pass)
    char buf[4];
    int* password_length = NULL;
    char password[MAX_PASSWORD_LENGTH];
+   char buffer[strlen(username) + 4];
```

Comment:
What is this 4 ?

Reply:
I am sending `strlen(username)` + `username` to the `remote` in the same stream so first 4 bytes are for the length of username string and the last `strlen(username)` bytes for the username data that's why I have allocated `strlen(username)+4` memory.